modules/dav/main/mod_dav.c

```diff
@@ -765,11 +779,27 @@ DAV_DECLARE(dav_error *) dav_get_resource(request_rec *r, int label_allowed,
                              ap_escape_html(r->pool, r->uri)));
     }
 
+    /* Take the repos root from DAVBasePath if configured, else the
+     * path of the enclosing section. */
+    base = conf->base ? conf->base : conf->dir;
+
     /* resolve the resource */
-    err = (*conf->provider->repos->get_resource)(r, conf->dir,
+    err = (*conf->provider->repos->get_resource)(r, base,
                                                  label, use_checked_in,
                                                  res_p);
     if (err != NULL) {
+        /* In the error path, give a hint that DavBasePath needs to be
+         * used if the location was configured via a regex match. */
+        if (!conf->base) {
+            core_dir_config *conf = ap_get_core_module_config(r->per_dir_config);
+
```

Comment:
Is it a good idea to use the same name `conf`for the `core_dir_config` variable as for the `dav_dir_conf` variable and thus mask it in the block?

Reply:
Definitely a bad idea! Fixed in 706af916933612750c1cf8f165e23d37c9b865b2